sdk/exportentities/workflow.go

```diff
@@ -468,70 +447,71 @@ func (w Workflow) GetWorkflow() (*sdk.Workflow, error) {
 		wf.Groups = append(wf.Groups, perm)
 	}
 
-	wf.Sort()
+	wf.SortNode()
 
 	return wf, nil
 }
 
-func (e *NodeEntry) getNode(name string) (*sdk.WorkflowNode, error) {
-	node := &sdk.WorkflowNode{
-		Name:         name,
-		Ref:          name,
-		PipelineName: e.PipelineName,
+func (e *NodeEntry) getNode(name string, proj *sdk.Project, w *sdk.Workflow) (*sdk.Node, error) {
+	node := &sdk.Node{
+		Name:    name,
+		Ref:     name,
+		Context: &sdk.NodeContext{},
 	}
 
-	if e.ApplicationName != "" {
-		node.Context = new(sdk.WorkflowNodeContext)
-		node.Context.Application = &sdk.Application{
-			Name: e.ApplicationName,
+	if e.PipelineName != "" {
+		for _, v := range proj.Pipelines {
+			if v.Name == e.PipelineName {
+				node.Context.PipelineID = v.ID
+				break
+			}
+		}
+		node.Type = sdk.NodeTypePipeline
+	} else {
+		if e.OutgoingHookModelName == "" {
+			node.Type = sdk.NodeTypeFork
 		}
 	}
 
-	if e.EnvironmentName != "" {
-		if node.Context == nil {
-			node.Context = new(sdk.WorkflowNodeContext)
+	if e.ApplicationName != "" {
```

Comment:
I think the new code if this workflow parser is to coupled with data coming from the database (ware are manipulating IDs).
you should only manipulate identify name and check them on API side, not in this sdk package
this was done with visitors in the workflow.Import function

Reply:
I didn't want to add fields: PipelineName, ApplicationName, HookModelName, EnvironmentName, OutgoingHookModelName  on the workflowdata struct to avoid duplicate (and useless) information.

Reply:
make the code more simpler and easier to tests is not useless :)

having to feed a whole project with applications, pipelines, environments, platforms all with corresponding IDs is a pain in the *ss for testing.
Another solution could be to have a map[string]Pipeline, map[string]Application, etc... rather than map[int64]Pipeline, map[int64]Application... for human readability it could be nice too